zerver/lib/bugdown/__init__.py

```diff
@@ -865,7 +876,13 @@ def run(self, root: Element) -> None:
             youtube = self.youtube_image(url)
             if youtube is not None:
                 yt_id = self.youtube_id(url)
-                add_a(root, youtube, url, None, None, "youtube-video message_inline_image", yt_id)
+                if self.get_youtube_title(url) is not None:
+                    yt_title = "YouTube - {}".format(self.get_youtube_title(url))
```

Comment:
Store the value of `self.get_youtube_title(url)` in a variable before the if. Otherwise, you are making two requests.